applications/forms.py

```diff
@@ -16,22 +16,52 @@ def __init__(self, *args, **kwargs):
         """
         The form here is programatically generated out of Question objects
         """
-
-        questions = kwargs.pop('questions')
+        self.form = kwargs.pop('form')
+        self.base_fields = generate_form_from_questions(
+            self.form.question_set.all())
+        self.base_fields.update({
+            'captcha': BetterReCaptchaField()
+        })
         super(ApplicationForm, self).__init__(*args, **kwargs)
-        self.fields = generate_form_from_questions(questions)
-        self.fields['captcha'] = BetterReCaptchaField()
+
+    def clean(self):
+        cleaned_data = self.cleaned_data
+        email = None
+        field_name = None
+
+        for name in self.cleaned_data:
+            question = None
+            pk = name.replace('question_', '')
+            value = self.cleaned_data[name]
+            try:
+                question = Question.objects.get(pk=pk, form=self.form)
+            except (Question.DoesNotExist, ValueError):
+                continue
```

Comment:
Additionally I think we should introduce some kind of checking if the form has just one e-mail field. Because currently if the form has two fields for e-mail (no idea what for, but events differ around the world) and the second one would be the applicant form... well, basically the wrong one would be attached to the `Application` object. But I have not idea how to change it at current stage so that it wouldn't take two days or something :D